src/toil/job.py

```diff
@@ -3758,6 +3826,181 @@ def getUserScript(self):
         return self.serviceModule
 
 
+class FileMetadata(NamedTuple):
+    """
+    Metadata for a file.
+    source is the URL to grab the file from
+    parent_dir is parent directory of the source
+    size is the size of the file. Is none if the filesize cannot be retrieved.
+    """
+
+    source: str
+    parent_dir: str
+    size: Optional[int]
+
+
+class CombineImportsJob(Job):
+    """
+    Combine the outputs of multiple WorkerImportsJob into one promise
+    """
+
+    def __init__(self, d: Sequence[Promised[Dict[str, FileID]]], **kwargs):
+        """
+        Merge the dicts
+        :param d: Sequence of dictionaries to merge
+        """
+        self._d = d
+        super().__init__(**kwargs)
+
+    def run(self, file_store: AbstractFileStore) -> Promised[Dict[str, FileID]]:
+        d = unwrap_all(self._d)
+        return {k: v for item in d for k, v in item.items()}
+
+
+class WorkerImportJob(Job):
+    """
+    Job to do file imports on a worker instead of a leader. Assumes all local and cloud files are accessible.
+
+    For the CWL/WDL runners, this class is only used when runImportsOnWorkers is enabled.
+    """
+
+    def __init__(
+        self,
+        filenames: List[str],
+        disk_size: Optional[ParseableIndivisibleResource] = None,
+        stream: bool = True,
+        **kwargs: Any,
+    ):
```

Comment:
Added a docstring. The logic the worker uses to do the import when streaming fails needs to know the amount of disk size it will use. This way, the command line won't report a failed job when streaming fails as I can catch it with a try except block and immediately reschedule. I'm not sure of a foolproof way to move setting the disk requirement to the caller. Then the caller would handle the job failing instead, and would report a failed job, which I don't want.